### PR TITLE
Elaborating error message when no startup is specified

### PR DESCRIPTION
**Pull request by JunTaoLuo**
#559

### CODE REVIEW

File: src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs [hunk 1]
```diff
@@ -43,7 +43,10 @@ public Type FindStartupType(string startupAssemblyName, IList<string> diagnostic
             var environmentName = _hostingEnv.EnvironmentName;
             if (string.IsNullOrEmpty(startupAssemblyName))
             {
-                throw new ArgumentException("Value cannot be null or empty.", nameof(startupAssemblyName));
+                throw new ArgumentException(
+                    string.Format("A startup method, startup type or startup assembly is required. If specifying an assembly, '{0}' cannot be null or empty.",
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs]: We don't have any resources?

File: src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs [hunk 2]
```diff
@@ -43,7 +43,10 @@ public Type FindStartupType(string startupAssemblyName, IList<string> diagnostic
             var environmentName = _hostingEnv.EnvironmentName;
             if (string.IsNullOrEmpty(startupAssemblyName))
             {
-                throw new ArgumentException("Value cannot be null or empty.", nameof(startupAssemblyName));
+                throw new ArgumentException(
+                    string.Format("A startup method, startup type or startup assembly is required. If specifying an assembly, '{0}' cannot be null or empty.", 
+                    nameof(startupAssemblyName)), 
+                    nameof(startupAssemblyName));
```
Comment (cesarbs) [inline src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs]: Why twice?

File: src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs (hunk 1 above)
Comment (JunTaoLuo) [inline src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs]: Not in hosting, I assume that will occur when localization strings are added down the road.

File: src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs (hunk 2 above)
Comment (JunTaoLuo) [inline src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs]: The first one is for the format string, the second is for the parameter name of the exception. I don't think the second one is that useful but it was there previously so I didn't remove it.

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**